roles/platform/tasks/initialize_setup_aws.yml

```diff
@@ -84,45 +84,49 @@
-
-    - name: Discover AWS VPC Public Subnets
-      when: not plat__aws_public_subnet_ids
-      block:
-        - name: Collect AWS Public Subnets
-          ansible.builtin.set_fact:
-            __aws_public_subnet_ids: "{{ __aws_public_subnet_ids | default([]) | union([__aws_subnet_item.subnet_id | default('')]) }}"
-          loop_control:
-            loop_var: __aws_subnet_item
-            label: "{{ __aws_subnet_item.subnet_id }}"
-          loop: "{{ __aws_subnets_info.subnets | selectattr('map_public_ip_on_launch') }}"
-
-        - name: Set fact for AWS Public Subnet IDs
-          ansible.builtin.set_fact:
-            plat__aws_public_subnet_ids: "{{ __aws_public_subnet_ids | default([]) }}"
-
-    - name: Set fact for AWS Private Subnet IDs if established by Infrastructure
-      when: not plat__aws_private_subnet_ids and infra__aws_private_subnet_ids is defined
+        plat__aws_public_subnet_ids: "{{ __aws_public_subnet_ids | default([]) }}"
+
+- name: Set fact for AWS Private Subnet IDs if established by Infrastructure
+  when:
+    - plat__aws_private_subnet_ids is undefined or not plat__aws_private_subnet_ids
+    - infra__aws_private_subnet_ids is defined
+  ansible.builtin.set_fact:
+    plat__aws_private_subnet_ids: "{{ infra__aws_private_subnet_ids }}"
+
+- name: Discover AWS VPC Private Subnets
+  when: plat__aws_private_subnet_ids is undefined or not plat__aws_private_subnet_ids
+  block:
+    - name: Collect AWS Private Subnets
+      ansible.builtin.set_fact:
+        __aws_private_subnet_ids: "{{ __aws_private_subnet_ids | default([]) | union([__aws_subnet_item.subnet_id | default('')]) }}"
+      loop_control:
+        loop_var: __aws_subnet_item
+        label: "{{ __aws_subnet_item.subnet_id }}"
+      loop: "{{ __aws_subnets_info.subnets | rejectattr('map_public_ip_on_launch') }}"
+
+    - name: Set fact for AWS Private Subnet IDs
       ansible.builtin.set_fact:
-        plat__aws_private_subnet_ids: "{{ infra__aws_private_subnet_ids }}"
-
-    - name: Discover AWS VPC Private Subnets
-      when: not plat__aws_private_subnet_ids
-      block:
-        - name: Collect AWS Private Subnets
-          ansible.builtin.set_fact:
-            __aws_private_subnet_ids: "{{ __aws_private_subnet_ids | default([]) | union([__aws_subnet_item.subnet_id | default('')]) }}"
-          loop_control:
-            loop_var: __aws_subnet_item
-            label: "{{ __aws_subnet_item.subnet_id }}"
-          loop: "{{ __aws_subnets_info.subnets | rejectattr('map_public_ip_on_launch') }}"
-
-        - name: Set fact for AWS Private Subnet IDs 
-          ansible.builtin.set_fact:
-            plat__aws_private_subnet_ids: "{{ __aws_private_subnet_ids | default([]) }}"
+        plat__aws_private_subnet_ids: "{{ __aws_private_subnet_ids | default([]) }}"
 
 # TODO: Discover AWS VPC Public Subnets if infra__ is not present
 - name: Set public subnets for public endpoint access
```

Comment:
Hey @Chaffelson - I think this task is no longer needed. You have handled `plat__aws_public_subnet_ids` in previous tasks, and also covered the discovery if infra__ isn't defined.

Reply:
I don't see the duplication - one is for public and one is for private, with the catcher at the bottom for when L0 it set?